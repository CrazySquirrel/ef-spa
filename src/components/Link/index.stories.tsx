import * as React from 'react';

import {storiesOf} from '@storybook/react';

import Link from './index';

storiesOf('Link', module)
    .add('current', () => (
        <div className='wrapper'>
            <Link>test link</Link>
        </div>
    ))
    .add('internal', () => (
        <div className='wrapper'>
            <Link to='/'>test link</Link>
        </div>
    ))
    .add('external', () => (
        <div className='wrapper'>
            <Link to='https://ef-spa.crazysquirrel.ru/'>test link</Link>
        </div>
    ));
