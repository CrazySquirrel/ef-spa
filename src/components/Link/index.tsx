declare const target: string;

import * as React from 'react';
import * as bem from 'bem-cn';

import * as cn from 'classnames';

import {prop} from '../../tools';

import {NavLink as RoutLink} from 'react-router-dom';

import {PAGES} from '../../routs';

import './index.scss';

export interface Props extends React.Props<Link> {
  to?: string;
  className?: string;
  replaceClassName?: string;
  title?: string;
}

const block = bem('link');

export default class Link extends React.Component<Props, {}> {
  public render() {
    const locals = {
      className: this.props.replaceClassName || cn(block(), this.props.className),
      title: this.props.title || prop(PAGES[this.props.to], 'metadata.menu'),
    };

    if (this.props.to) {
      if (this.props.to.startsWith('http') || target === 'storybook') {
        return (
            <a
                {...locals}
                target='_blank'
                rel='noreferrer'
                href={this.props.to}
            >
              {this.props.children}
            </a>
        );
      } else {
        return (
            <RoutLink
                {...locals}
                to={this.props.to}
                exact={true}
                activeClassName={this.getActiveClassName()}
            >
              {this.props.children}
            </RoutLink>
        );
      }
    } else {
      return (
          <span {...locals}>
            {this.props.children}
          </span>
      );
    }
  }

  private getActiveClassName() {
    // istanbul ignore next
    return this.props.replaceClassName || block({active: true})();
  }
}
