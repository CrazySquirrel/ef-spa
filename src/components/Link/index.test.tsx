import * as React from 'react';
import {shallow} from 'enzyme';

import Link from './index';

describe('Link', () => {
  it('render', () => {
    const result = shallow(<Link/>);

    expect(result).toMatchSnapshot();
  });
});
