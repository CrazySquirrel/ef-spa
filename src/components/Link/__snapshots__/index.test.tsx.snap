version https://git-lfs.github.com/spec/v1
oid sha256:dedd34750c779ef8968476c3ca1e9094edb1345abd81c91974eddf2fb2564787
size 12735
